// @api-platform/admin/lib/layout/AppBar.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _reactAdmin = require("react-admin");

var _react = _interopRequireDefault(require("react"));

var _Typography = _interopRequireDefault(require("@material-ui/core/Typography"));

var _styles = require("@material-ui/core/styles");

var _Logo = _interopRequireDefault(require("./Logo"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const styles = {
  title: {
    flex: 1,
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',
    overflow: 'hidden'
  },
  spacer: {
    flex: 1
  }
};
const CustomAppBar = (0, _styles.withStyles)(styles)(({
  classes,
  ...props
}) => _react.default.createElement(_reactAdmin.AppBar, props, _react.default.createElement(_Typography.default, {
  variant: "title",
  color: "inherit",
  className: classes.title,
  id: "react-admin-title"
}), _react.default.createElement(_Logo.default, null), _react.default.createElement("span", {
  className: classes.spacer
})));
var _default = CustomAppBar;
exports.default = _default;
