// @api-platform/admin/lib/EditGuesser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.EditGuesserComponent = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactAdmin = require("react-admin");

var _InputGuesser = _interopRequireDefault(require("./InputGuesser"));

var _IntrospectQuery = _interopRequireDefault(require("./IntrospectQuery"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const displayOverrideCode = (resourceSchema, fields) => {
  let code = 'If you want to override at least one input, paste this content in the <EditGuesser> component of your resource:\n\n';
  code += `const ${resourceSchema.title}Edit = props => (\n`;
  code += `    <EditGuesser {...props}>\n`;
  fields.forEach(field => {
    code += `        <InputGuesser source={"${field.name}"} />\n`;
  });
  code += `    </EditGuesser>\n`;
  code += `);\n`;
  code += `\n`;
  code += `And don't forget update your <ResourceGuesser> component:\n`;
  code += `<ResourceGuesser name={"${resourceSchema.name}"} edit={${resourceSchema.title}Edit} />`;
  console.info(code);
}; // useful for testing (we don't need Query)


const EditGuesserComponent = ({
  children,
  fields,
  resourceSchema,
  ...props
}) => {
  if (!children) {
    children = fields.map(field => _react.default.createElement(_InputGuesser.default, {
      key: field.name,
      source: field.name
    }));
    displayOverrideCode(resourceSchema, fields);
  }

  return (// FIXME: enabling the undoable feature breaks API Platform Admin...
    // see https://github.com/api-platform/admin/pull/217
    _react.default.createElement(_reactAdmin.Edit, _extends({}, props, {
      undoable: false
    }), _react.default.createElement(_reactAdmin.SimpleForm, null, children))
  );
};

exports.EditGuesserComponent = EditGuesserComponent;

const EditGuesser = props => _react.default.createElement(_IntrospectQuery.default, _extends({
  component: EditGuesserComponent
}, props));

EditGuesser.propTypes = {
  children: _propTypes.default.oneOfType([_propTypes.default.node, _propTypes.default.func]),
  resource: _propTypes.default.string.isRequired
};
var _default = EditGuesser;
exports.default = _default;
