// @api-platform/admin/lib/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
var _exportNames = {
  CreateGuesser: true,
  EditGuesser: true,
  FieldGuesser: true,
  InputGuesser: true,
  ListGuesser: true,
  ResourceGuesser: true,
  ShowGuesser: true
};
Object.defineProperty(exports, "CreateGuesser", {
  enumerable: true,
  get: function () {
    return _CreateGuesser2.default;
  }
});
Object.defineProperty(exports, "EditGuesser", {
  enumerable: true,
  get: function () {
    return _EditGuesser2.default;
  }
});
Object.defineProperty(exports, "FieldGuesser", {
  enumerable: true,
  get: function () {
    return _FieldGuesser2.default;
  }
});
Object.defineProperty(exports, "InputGuesser", {
  enumerable: true,
  get: function () {
    return _InputGuesser2.default;
  }
});
Object.defineProperty(exports, "ListGuesser", {
  enumerable: true,
  get: function () {
    return _ListGuesser2.default;
  }
});
Object.defineProperty(exports, "ResourceGuesser", {
  enumerable: true,
  get: function () {
    return _ResourceGuesser2.default;
  }
});
Object.defineProperty(exports, "ShowGuesser", {
  enumerable: true,
  get: function () {
    return _ShowGuesser2.default;
  }
});

var _CreateGuesser2 = _interopRequireDefault(require("./CreateGuesser"));

var _EditGuesser2 = _interopRequireDefault(require("./EditGuesser"));

var _FieldGuesser2 = _interopRequireDefault(require("./FieldGuesser"));

var _InputGuesser2 = _interopRequireDefault(require("./InputGuesser"));

var _ListGuesser2 = _interopRequireDefault(require("./ListGuesser"));

var _ResourceGuesser2 = _interopRequireDefault(require("./ResourceGuesser"));

var _ShowGuesser2 = _interopRequireDefault(require("./ShowGuesser"));

var _hydra = require("./hydra");

Object.keys(_hydra).forEach(function (key) {
  if (key === "default" || key === "__esModule") return;
  if (Object.prototype.hasOwnProperty.call(_exportNames, key)) return;
  Object.defineProperty(exports, key, {
    enumerable: true,
    get: function () {
      return _hydra[key];
    }
  });
});

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
