// @api-platform/admin/lib/InputGuesser.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.InputGuesserComponent = void 0;

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactAdmin = require("react-admin");

var _docsUtils = require("./docsUtils");

var _IntrospectQuery = _interopRequireDefault(require("./IntrospectQuery"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function () { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const InputGuesserComponent = ({
  fields,
  resourceSchema,
  ...props
}) => {
  const field = fields.find(({
    name
  }) => name === props.source);

  if (!field) {
    console.error(`Field ${props.source} not present inside the api description for the resource ${props.resource}`);
    return _react.default.createElement(_react.Fragment, null);
  }

  const validate = !props.validate && field.required ? [(0, _reactAdmin.required)()] : props.validate;

  if (null !== field.reference) {
    if (1 === field.maxCardinality) {
      return _react.default.createElement(_reactAdmin.ReferenceInput, _extends({
        key: field.name,
        label: field.name,
        reference: field.reference.name,
        source: field.name,
        validate: validate
      }, props, {
        allowEmpty: true
      }), _react.default.createElement(_reactAdmin.SelectInput, {
        optionText: (0, _docsUtils.getReferenceNameField)(field.reference)
      }));
    }

    return _react.default.createElement(_reactAdmin.ReferenceArrayInput, _extends({
      key: field.name,
      label: field.name,
      reference: field.reference.name,
      source: field.name,
      validate: validate
    }, props, {
      allowEmpty: true
    }), _react.default.createElement(_reactAdmin.SelectArrayInput, {
      optionText: (0, _docsUtils.getReferenceNameField)(field.reference)
    }));
  }

  if ('http://schema.org/identifier' === field.id) {
    const prefix = `/${props.resource}/`;

    props.format = value => {
      return value && 0 === value.indexOf(prefix) ? value.substr(prefix.length) : value;
    };

    props.parse = value => {
      return -1 !== value.indexOf(prefix) ? prefix + value : value;
    };
  }

  switch (field.range) {
    case 'http://www.w3.org/2001/XMLSchema#array':
      return _react.default.createElement(_reactAdmin.ArrayInput, _extends({
        key: field.name,
        source: field.name,
        validate: validate
      }, props), _react.default.createElement(_reactAdmin.SimpleFormIterator, null, _react.default.createElement(_reactAdmin.TextInput, null)));

    case 'http://www.w3.org/2001/XMLSchema#integer':
      return _react.default.createElement(_reactAdmin.NumberInput, _extends({
        key: field.name,
        source: field.name,
        validate: validate
      }, props));

    case 'http://www.w3.org/2001/XMLSchema#decimal':
      return _react.default.createElement(_reactAdmin.NumberInput, _extends({
        key: field.name,
        source: field.name,
        step: "0.1",
        validate: validate
      }, props));

    case 'http://www.w3.org/2001/XMLSchema#boolean':
      return _react.default.createElement(_reactAdmin.BooleanInput, _extends({
        key: field.name,
        source: field.name,
        validate: validate
      }, props));

    case 'http://www.w3.org/2001/XMLSchema#date':
    case 'http://www.w3.org/2001/XMLSchema#dateTime':
      return _react.default.createElement(_reactAdmin.DateInput, _extends({
        key: field.name,
        source: field.name,
        validate: validate
      }, props));

    default:
      return _react.default.createElement(_reactAdmin.TextInput, _extends({
        key: field.name,
        source: field.name,
        validate: validate
      }, props));
  }
};

exports.InputGuesserComponent = InputGuesserComponent;

const InputGuesser = props => _react.default.createElement(_IntrospectQuery.default, _extends({
  component: InputGuesserComponent,
  includeDeprecated: true
}, props));

InputGuesser.propTypes = {
  source: _propTypes.default.string.isRequired
};
var _default = InputGuesser;
exports.default = _default;
