// @api-platform/admin/lib/hydra/HydraAdmin.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _reactRedux = require("react-redux");

var _history = require("history");

var _withContext = _interopRequireDefault(require("recompose/withContext"));

var _reactAdmin = require("react-admin");

var _dataProvider = _interopRequireDefault(require("./dataProvider"));

var _AdminGuesser = _interopRequireDefault(require("../AdminGuesser"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const history = (0, _history.createHashHistory)();

const HydraAdmin = ({
  entrypoint,
  dataProvider = (0, _dataProvider.default)(entrypoint),
  i18nProvider,
  authProvider,
  customReducers,
  customSagas,
  locale,
  ...rest
}) => _react.default.createElement(_reactRedux.Provider, {
  store: (0, _reactAdmin.createAdminStore)({
    authProvider,
    dataProvider,
    i18nProvider,
    history,
    customReducers,
    customSagas,
    locale
  })
}, _react.default.createElement(_AdminGuesser.default, _extends({
  dataProvider: dataProvider,
  authProvider: authProvider,
  history: history,
  i18nProvider: i18nProvider,
  customReducers: customReducers,
  customSagas: customSagas,
  locale: locale
}, rest)));

var _default = (0, _withContext.default)({
  authProvider: _propTypes.default.func
}, props => ({
  authProvider: props.authProvider
}))(HydraAdmin);

exports.default = _default;
