// @api-platform/admin/lib/common-test/common-data-test.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.API_FIELDS_DATA = exports.API_INPUT_DATA = exports.RESOURCE_NAME = exports.ENTRYPOINT = void 0;

var _Field = _interopRequireDefault(require("@api-platform/api-doc-parser/lib/Field"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const ENTRYPOINT = 'http://entrypoint';
exports.ENTRYPOINT = ENTRYPOINT;
const RESOURCE_NAME = 'user';
exports.RESOURCE_NAME = RESOURCE_NAME;
const API_INPUT_DATA = {
  resources: [{
    name: `${RESOURCE_NAME}`,
    fields: [new _Field.default('fieldA', {
      id: 'http://schema.org/fieldA',
      range: 'http://www.w3.org/2001/XMLSchema#string',
      reference: null,
      required: true
    }), new _Field.default('fieldB', {
      id: 'http://schema.org/fieldB',
      range: 'http://www.w3.org/2001/XMLSchema#string',
      reference: null,
      required: true
    }), new _Field.default('deprecatedField', {
      id: 'http://localhost/deprecatedField',
      range: 'http://www.w3.org/2001/XMLSchema#string',
      reference: null,
      required: true,
      deprecated: true
    })],
    parameters: [],
    url: `${ENTRYPOINT}/users`
  }]
};
exports.API_INPUT_DATA = API_INPUT_DATA;
const API_FIELDS_DATA = [new _Field.default('fieldA', {
  id: 'http://schema.org/fieldA',
  range: 'http://www.w3.org/2001/XMLSchema#string',
  reference: null,
  required: true
}), new _Field.default('fieldB', {
  id: 'http://schema.org/fieldB',
  range: 'http://www.w3.org/2001/XMLSchema#string',
  reference: null,
  required: true
}), new _Field.default('deprecatedField', {
  id: 'http://localhost/deprecatedField',
  range: 'http://www.w3.org/2001/XMLSchema#string',
  reference: null,
  required: true,
  deprecated: true
})];
exports.API_FIELDS_DATA = API_FIELDS_DATA;
